src/pam/pam_oslogin_admin.cc

```diff
@@ -0,0 +1,47 @@
+// Copyright 2024 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <security/pam_modules.h>
+
+#include <compat.h>
+#include <oslogin_utils.h>
+
+using std::string;
+
+using oslogin_utils::AuthOptions;
+
+extern "C" {
+
+PAM_EXTERN int
+pam_sm_acct_mgmt(pam_handle_t* pamh, int flags, int argc, const char** argv) {
+  struct AuthOptions opts;
+  const char *user_name;
+  string user_response;
+
+  if (pam_get_user(pamh, &user_name, NULL) != PAM_SUCCESS) {
+    PAM_SYSLOG(pamh, LOG_INFO, "Could not get pam user.");
```

Comment:
Please include the name of the user who could not be found in this error message.

Reply:
I this case it'll not be possible as we are logging that we couldn't get it.

Reply:
I meant just the value of `user_name` which couldn't be looked up. Something like:

    PAM_SYSLOG(pamh, LOG_INFO, "Could not get pam user %s.", user_name);

Reply:
we don't have `user_name` at this point, the call to `pam_get_user()` failed, that means we can't use whatever `user_name` is pointing to - it's not the actual user name.

Reply:
Ahhh yep, I'm bad at C++: misread user_name as an input parameter, rather than an output param. Sounds good.